saleor/cart/utils.py

```diff
@@ -51,3 +53,12 @@ def check_product_availability_and_warn(request, cart):
                 'Quantity was set to maximum available for now.')
         messages.warning(request, msg)
         remove_unavailable_variants(cart)
+
+
+def process_purchase(request, cart, product):
```

Comment:
What name do you suggest? `process_add_to_cart` or `process_cart_form` would be better?